### ISSUE TITLE
Physical unit of Thermal.HeatTransfer.Celsius.TemperatureSensor.T

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#830]
**Reported by anonymous on 19 Sep 2012 11:48 UTC**
Should not better `Modelica.Thermal.HeatTransfer.Celsius.TemperatureSensor.T` be of type `Modelica.SIunits.Conversions.NonSIunits.Temperature_degC` instead of type `Modelica.Blocks.Interfaces.RealOutput`? Otherwise I wonder how Dymola 2012 FD01 Preview can display the `degC` unit in the plot window.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/830

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 19 Sep 2012 12:09 UTC**
In the same context `Modelica.Thermal.HeatTransfer.Sensors.TemperatureSensor.T` should be of type `Modelica.SIunits.Temperature`.

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 28 Sep 2012 07:58 UTC**
Guessing the physical unit in Dymola 2013 FD01 Preview is misleading as you can see from test models like

```
model TemperatureSensor1
 Modelica.Blocks.Interfaces.RealOutput T annotation(Placement(transformation(extent={{90,-10},{110,10}})));
 Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a port annotation(Placement(transformation(extent={{-110,-10},{-90,10}})));
equation
 T = Modelica.SIunits.Conversions.to_degC(port.T);
 port.Q_flow = 0;
end TemperatureSensor1;

model TemperatureSensor2
 Modelica.Blocks.Interfaces.RealOutput T annotation(Placement(transformation(extent={{90,-10},{110,10}})));
 Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a port annotation(Placement(transformation(extent={{-110,-10},{-90,10}})));
equation
 T = port.T + Modelica.Constants.T_zero;
 port.Q_flow = 0;
end TemperatureSensor2;

model TemperatureSensor3
 Modelica.SIunits.Conversions.NonSIunits.Temperature_degC T annotation(Placement(transformation(extent={{90,-10},{110,10}})));
 Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a port annotation(Placement(transformation(extent={{-110,-10},{-90,10}})));
equation
 T = Modelica.SIunits.Conversions.to_degC(port.T);
 port.Q_flow = 0;
end TemperatureSensor3;

model TemperatureSensor4
 Modelica.SIunits.Conversions.NonSIunits.Temperature_degC T annotation(Placement(transformation(extent={{90,-10},{110,10}})));
 Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a port annotation(Placement(transformation(extent={{-110,-10},{-90,10}})));
equation
 T = port.T + Modelica.Constants.T_zero;
 port.Q_flow = 0;
end TemperatureSensor4;
```

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 15 Oct 2012 07:28 UTC**
Please set milestone to MSL3.2.1. Otherwise it is mising on the roadmap.

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 15 Oct 2012 10:45 UTC**
See also #876.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 4 Dec 2012 14:16 UTC**
This issue was closed as part of fixing #876 in f2f3947370f7cfdee9c0a0c20a1bddd0b626247e.
